### PR TITLE
Added codeclimate configuration

### PR DESCRIPTION
**Pull request by MarioBlazek**
@andrerom  This PR continues where https://github.com/ezsystems/ezpublish-kernel/pull/2111 has left off.

### CODE REVIEW

File: .codeclimate.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    - "eZ/Publish/SPI/Tests/"
+
+# PHP code anlysis engine
+# PHPCodeSniffer, PHPMD and CodeClimate duplication engine
+engines:
+    phpcodesniffer:
+        enabled: true
+        config:
+            file_extensions: "php"
+            standard: "PSR1,PSR2"
```
Comment (andrerom) [inline .codeclimate.yml]: Question from @alongosz: Do we need this? Does it provide any value as we already have own cs service using php-cs-fixer config for CS?

File: .codeclimate.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    - "eZ/Publish/Core/Persistence/Doctrine/Tests/"
+    - "eZ/Publish/Core/Persistence/Legacy/Tests/"
+    - "eZ/Publish/Core/Repository/Tests/"
+    - "eZ/Publish/Core/REST/Client/Tests/"
+    - "eZ/Publish/Core/REST/Common/Tests/"
+    - "eZ/Publish/Core/REST/Server/Tests/"
+    - "eZ/Publish/Core/Search/Tests/"
+    - "eZ/Publish/Core/Search/Legacy/Tests/"
+    - "eZ/Publish/Core/SignalSlot/Tests/"
+    - "eZ/Publish/SPI/Tests/"
```
Comment (andrerom) [inline .codeclimate.yml]: can these user pattern matches instead for Tests folders?

File: .codeclimate.yml (hunk 1 above)
Comment (MarioBlazek) [reply .codeclimate.yml]: @andrerom regarding your question, this config is redundant in that case.

File: .codeclimate.yml (hunk 2 above)
Comment (MarioBlazek) [reply .codeclimate.yml]: @andrerom not sure what you mean

Comment (andrerom) [reply .codeclimate.yml]: Like here: https://docs.codeclimate.com/v1.0/docs/example-codeclimateyml

Comment (andrerom) [reply .codeclimate.yml]: As in can we match something like "**/Tests/" ?

File: .codeclimate.yml (hunk 1 above)
Comment (andrerom) [reply .codeclimate.yml]: Ok, lets remove it then as it's actually more relaxed than wheat we use.
If this service would support php-cs-fixer we could enable that.

**Comment by MarioBlazek** [comment]
@andrerom to sum up. I've disabled phpcodesniffer and added Tests exclude pattern from documentation.

**Comment by MarioBlazek** [comment]
I think that in open source version only one branch can be monitored.

**Comment by alongosz** [comment]
> I think that in open source version only one branch can be monitored.

Hmm, that's kind of huge limitation of the tool as it leads to an unwanted side effect when you're backporting bug and merging up to `master`. Any CodeClimate error then would require special handling after merging (we rather don't PR merged backports to master, we just merge branches up).

But anyway, can we switch tested branch temporarily so we're sure at least for `6.7` it does not fail?

**Comment by MarioBlazek** [comment]
@alongosz previous PR was open against master.

**Comment by alongosz** [comment]
Hmm, @MarioBlazek I tried to integrate CodeClimate with my fork and got the following error for config:
https://codeclimate.com/github/alongosz/ezpublish-kernel/pull/3
Relevant cherry-picked test PR: https://github.com/alongosz/ezpublish-kernel/pull/3/files

This is why I'm against merging this until we can see it (final version) passes tests. Some build that passed in the past might be no longer relevant.

Also if it's possible to test only for PRs on master, IMHO it's a show-stopper, because failing CI makes you ignore it and merge code that has actual errors.

**Comment by MarioBlazek** [comment]
@alongosz I've also did the same on my fork. It looks like they upgraded codeclimate to new version. Same config works fine on my repos. I will do some research and update PR accordingly.

**Comment by MarioBlazek** [comment]
@alongosz just to confirm, codeclimate does not support PR on any other branch than master (or default branch). Currently there is no a lot of tools which do enable tracking PR's on other branches (in open source model).

**Comment by alongosz** [comment]
> @alongosz just to confirm, codeclimate does not support PR on any other branch than master (or default branch). Currently there is no a lot of tools which do enable tracking PR's on other branches (in open source model).

Well... Travis does that very well for us :)
The implication of that limitation is necessity of running `codeclimate` manually at least when doing code review (beside `master`, we're supporting `6.7`, `6.12`, `7.0` right now).

I'd say the tool is not best suited for our needs, but leaving decision on how we'll handle non-master PRs to @andrerom (since you both consulted on this for a long time now).

**Comment by andrerom** [comment]
How does scrutinizer-ci or codacy compare? _(or others)_

**Comment by andrerom** [comment]
@alongosz That said, codeclimate and similar is maybe most useful for new features and to see trends to sometime act upon them. So it might be ok that this just runs on master.

**Comment by alongosz** [comment]
Makes sense, but let's introduce it on `master` then :) 
Otherwise we'll end up with code that fails after merging up (*not that this doesn't happen, I know ;)*)
But if you really want this on `6.7` then let's at least switch temporarily, in CodeClimate settings, default branch to `6.7` to see if it works at all.

**Comment by andrerom** [comment]
ok, master it is. but seems it does not pass @MarioBlazek

**Comment by MarioBlazek** [comment]
@andrerom @alongosz will check, thanks.

**Comment by andrerom** [comment]
Note: Disabled code climate webhook as it was starting to get annoying that it fails.

**Comment by alongosz** [comment]
Seems like nothing changed in Code Climate over last 1.5 year. It still supports master builds only AFAICS (maybe it's just for OSS/free plan).

I'm sure it's a great tool, but it's not suitable for our development workflow. I'm sorry to close this one.

**Pull request closed without merging**